firebase-firestore/src/main/java/com/google/firebase/firestore/bundle/BundleSerializer.java

```diff
@@ -38,16 +38,35 @@
 import com.google.protobuf.ByteString;
 import com.google.protobuf.NullValue;
 import com.google.type.LatLng;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
 import java.util.ArrayList;
+import java.util.Date;
+import java.util.GregorianCalendar;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Locale;
+import java.util.TimeZone;
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
 
 /** A JSON serializer to deserialize Firestore Bundles. */
 class BundleSerializer {
 
+  private static final long MILLIS_PER_SECOND = 1000;
+
+  private static final SimpleDateFormat timestampFormat;
```

Comment:
DateFormat objects are not safe to use from multiple threads. While any one Firestore instance may only use this from a single thread, multiple Firestore instances can run at the same time. You could make this safe by making this a member rather than a static.

Reply:
I thought this problem didn't apply to us since we are single-threaded, but you are right that it shows up when multiple instances are used. Thanks for pointing this out.